### DISCUSSION TITLE
Read in/out bytes in updown script

### DISCUSSION THREAD

**Discussion by realbitt** [strongswan/strongswan#818] [General]
Is there any way to read  in/out bytes traffic for  ip,user or connection inside updown script ?

**Comment by tobiasbrunner** [comment]
No. The [vici](https://wiki.strongswan.org/projects/strongswan/wiki/vici) _child-updown_ and _child-rekey_ events do provide stats for the affected CHILD_SAs, though (at least if they are available, which is usually the case as SAs are generally rekeyed before they expire). Note that this only provides stats for individual CHILD_SAs. So you might want to consider [RADIUS accounting](https://wiki.strongswan.org/projects/strongswan/wiki/EapRadius#Accounting), which provides cumulative stats per IKE_SA (does not require authentication via RADIUS, see [here](https://wiki.strongswan.org/projects/strongswan/wiki/EapRadius#Attributes-sent-to-RADIUS-servers) for the attributes sent in the Accounting messages).

**Reply by realbitt** [reply]
I will read about  RADIUS accounting and I'll make some tests.
Thanks @tobiasbrunner
